scripts/create-secret.sh

```diff
@@ -104,69 +91,74 @@ function validate_length() {
   # Validate minimum length
   if [[ $1 =~ $ALPHANUMERIC_DASH$MIN_LENGTH ]]
   then
-    echo $GREEN'Length: Ok!'$RESET
+    printf "%sLength: Ok! %s\n" "$GREEN" "$RESET"
   else
-    echo $RED'To short, should be 6-256 characters long.'$RESET
+    printf "%sTo short, should be 6-256 characters long.%s\n" "$RED" "$RESET"
     exit 0
   fi
 }
 
 #-------------------CREATE SECRET--------------------------#
-function prepare_secret () {
+prepare_secret () {
   # Prompt user for secret name
-  read -p $BLUE"Secret name: $RESET$SSM_PREFIX" SECRET_NAME
+  read -erp "${BLUE}Secret name: ${RESET}${SSM_PREFIX}" SECRET_NAME
   validate_whitespace "$SECRET_NAME"
   validate_chars "$SECRET_NAME"
   validate_length "$SECRET_NAME"
 
   # Prompt user for secret value
-  read -p $BLUE'Secret value: '$RESET SECRET_VALUE
+  read -erp "${BLUE}Secret value: ${RESET}" SECRET_VALUE
   validate_whitespace "$SECRET_VALUE"
   validate_length "$SECRET_VALUE"
 
+  # Prompt user for secret type
+  read -erp "${BLUE}SecureString [Y/n]? ${RESET}"
+  if [[ $REPLY =~ ^[Nn]$ ]]
+  then
+    SECRET_TYPE="String"
+  else
+    SECRET_TYPE="SecureString"
+  fi
+  printf "%s$SECRET_TYPE selected%s\n" "$GREEN" "$RESET"
 
-  read -p $YELLOW"Are you sure [y/n]? "$RESET -r
+  # Prompt user for adding tags
+  TAGS=""
+  read -erp "${BLUE}Add tags? [y/N]? ${RESET}"
   if [[ $REPLY =~ ^[Yy]$ ]]
   then
-    echo # newline
-    echo $GREEN"Creating secret...."$RESET
-    create_secret $SECRET_NAME $SECRET_VALUE
+    read -erp "${YELLOW}Example: Key=Foo,Value=Bar Key=Another,Value=Tag: ${RESET}" TAGS
+  fi
+  read -erp "${YELLOW}Are you sure [Y/n]? ${RESET}"
+  if [[ $REPLY =~ ^[Nn]$ ]]
+  then
+    printf "%sAborting...%s" "$RED" "$RESET"
   else
-    echo $RED"Aborting..."$RESET
+    printf "%sCreating secret....%s\n" "$GREEN" "$RESET"
+    create_secret "$SECRET_NAME" "$SECRET_VALUE" "$SECRET_TYPE" "$TAGS"
   fi
 }
 
-function create_secret () {
+create_secret () {
   SECRET_NAME=$1
   SECRET_VALUE=$2
-  aws ssm put-parameter --name $SSM_PREFIX$SECRET_NAME --value $SECRET_VALUE --type SecureString
-  echo $GREEN"Done!"$RESET
+  SECRET_TYPE=$3
+  TAGS=$4
+  CMD="aws ssm put-parameter --name $SSM_PREFIX$SECRET_NAME --value $SECRET_VALUE --type $SECRET_TYPE"
+
+  if [ -n "$TAGS" ]
+  then
+    CMD="$CMD --tags $TAGS"
+  fi
+  eval "$CMD"
+  printf "%sDone!%s" "$GREEN" "$RESET"
 }
 
+case $1 in
```

Comment:
I don't get this case. Is this for testing purposes? Maybe add a comment about it then?

Reply:
used in [shellspec](https://shellspec.info/) but threw shellspec out. Forgot to clean up this line.